polaris-react/src/components/SearchableListbox/SearchableListbox.tsx

```diff
@@ -0,0 +1,120 @@
+import {Scrollable} from '../Scrollable';
+import {AutoSelection, Listbox} from '../Listbox';
+import {useUniqueId} from '../../utilities/unique-id';
+
+import {StopPropagation, Search, SearchEmptyState} from './components';
+import styles from './SearchableListbox.scss';
+
+export interface SearchableListItem {
+  children: string | ReactNode;
+  value: string;
+  selected?: boolean;
+}
+
+export interface SearchableListboxProps {
+  activatorNode: ReactElement;
+  open: boolean;
+  showSearch: boolean;
+  searchValue: string;
+  searchEmptyStateMessage: string;
+  searchLabel: string;
+  searchPlaceholder?: string;
+  listItems?: SearchableListItem[];
+  loading?: boolean;
+  loadingAccessibilityLabel?: string;
+  footerAction?: ReactElement;
+  onClose(): void;
+  onSearch(search: string): void;
+  onOptionSelect(selected: string): void;
+  onScrolledToBottom?(): void;
+}
+
+export function SearchableListbox({
+  activatorNode,
+  open,
+  showSearch,
+  searchValue,
+  searchLabel,
+  searchPlaceholder,
+  searchEmptyStateMessage,
+  loading,
+  loadingAccessibilityLabel,
+  listItems,
+  footerAction,
+  onClose,
+  onOptionSelect,
+  onSearch,
+  onScrolledToBottom,
+}: SearchableListboxProps) {
+  const [activeOptionDomId, setActiveOptionDomId] = useState<string>();
+  const listId = useUniqueId('SearchableListbox');
+
+  const showEmptyState = !loading && !listItems?.length;
+
+  const handleActiveOptionChange = (_: string, domId: string) => {
+    setActiveOptionDomId(domId);
+  };
+
+  return (
+    <Popover
+      preferredAlignment="left"
+      activator={activatorNode}
+      active={open}
+      onClose={onClose}
+    >
+      {showSearch && (
+        <Popover.Pane fixed>
+          <div className={styles.SearchFieldWrapper}>
+            <StopPropagation>
+              <Search
+                activeOptionDomId={activeOptionDomId}
+                label={searchLabel}
+                value={searchValue}
+                placeholder={searchPlaceholder}
+                listId={listId}
+                onSearch={onSearch}
+              />
+            </StopPropagation>
+          </div>
+        </Popover.Pane>
+      )}
+      {showEmptyState ? (
+        <SearchEmptyState message={searchEmptyStateMessage} />
+      ) : (
+        <Scrollable shadow onScrolledToBottom={onScrolledToBottom}>
+          <div className={styles.ListBoxWrapper}>
+            <Listbox
+              autoSelection={AutoSelection.None}
+              onSelect={onOptionSelect}
+              customListId={listId}
+              onActiveOptionChange={handleActiveOptionChange}
+            >
+              {listItems?.map(({children, value, selected}) => {
+                return (
+                  <Listbox.Option value={value} selected={selected} key={value}>
+                    <Listbox.TextOption selected={selected}>
+                      {children}
+                    </Listbox.TextOption>
+                  </Listbox.Option>
```

Comment:
> For example in the LocationPicker we use a Listbox.Action for the Show all locations action.

We were initially using the Listbox.Actio, but it didn't entirely match the styles from our original figmas iirc, but happy to use it now. It would indeed fix this double focus state. I feel like having a shared style for the selected states Option > TextOption could be useful, but I agree that it is limiting. It depends on how extensible we want this component to be or if it should follow the base UI for the [pattern](https://polaris.shopify.com/foundations/patterns/locations) and evolve with it 🤔

Reply:
@felipeleusin @dmcvittie as consumers, how do you feel about the point above regarding having consumers fully control what type of `Listbox.[SubComponent]` are used to render each list item? initially, on the segmentation side, we wanted to provide a plug (data) & play, consistent UI to y'all so that's our rationale. 

@andrewmcgov do you foresee a lot of use cases where you would not use `Listbox.Option` > `Listbox.TextOption`? it looked from the location picker mockups that this would be alright. i will definitely change the footer action to be a `Listbox.Action` of course

Reply:
I lean towards having consumers control the rendering of the items because it allows for full flexibility of appearance and order of items. 

That being said, if the footer action is rendered as a `Listbox.Action` I am fine either way as it supports our use case!

Reply:
I wouldn't be too opposed to build the components using `Listbox.Option` directly if it makes us with a bit more flexible composition

Reply:
just so we are all clear on it, @felipeleusin @dmcvittie, this would mean that you would be doing this yourselves:

```
 {listItems?.map(({children, value, selected}) => {
                return (
                  <Listbox.Option value={value} selected={selected} key={value}>
                    <Listbox.TextOption selected={selected}>
                      {children}
                    </Listbox.TextOption>
                  </Listbox.Option>
                );
              })}